lib/versioning/pvp/index.ts

```diff
@@ -0,0 +1,84 @@
+import { regEx } from '../../util/regex';
+import { GenericVersion, GenericVersioningApi } from '../generic';
+import type { VersioningApi } from '../types';
+
+export const id = 'pvp';
+export const displayName = 'PVP';
+export const urls = ['https://pvp.haskell.org'];
+export const supportsRanges = false;
+
+/**
+ * At least three components and no leading zeros
+ * https://pvp.haskell.org/#version-number
+ */
+const versionPattern = regEx(
+  /^((([0-9])|([1-9][0-9]*))\.){2,}(([0-9])|([1-9][0-9]*))$/
+);
+
+class PvpVersioningApi extends GenericVersioningApi {
+  /**
+   * PVP has two major components A.B
+   * To keep compatibility with Renovate's versioning API we will treat it as a float
+   */
+  protected _parse(version: string): GenericVersion {
+    const matches = versionPattern.exec(version);
+    if (!matches) {
+      return null;
+    }
+
+    const components = version.split('.');
+    const major = parseFloat(`${components[0]}.${components[1]}`);
+    const rest = components.slice(2).map((i) => parseInt(i));
```

Comment:
```suggestion
    const rest = components.slice(2).map((i) => parseInt(i, 10));
```
Parse as base10, better safe practice.